File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    lost.sort()
    reserve.sort()
    reserve_filter = [r for r in reserve if r not in lost]
    lost_filter = [l for l in lost if l not in reserve]

    for r in reserve_filter:
        temp1 = r - 1
        temp2 = r + 1
        if temp1 in lost_filter:
            lost_filter.remove(temp1)
        elif temp2 in lost_filter:
            lost_filter.remove(temp2)
    return n - len(lost_filter)

File: 프로그래머스/1/42862. 체육복/체육복.swift
import Foundation


func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    var lostSet = Set(lost).subtracting(reserve)
    let reserveSet = Set(reserve).subtracting(lost).sorted(by: <)
    
    for reserve in reserveSet {
        if lostSet.contains(reserve - 1) {      
            lostSet.remove(reserve - 1)
            continue
        }
        if lostSet.contains(reserve + 1) {    
            lostSet.remove(reserve + 1)
        }
    }
    return (n - lostSet.count)
}
